### PR TITLE
Added Particle Emitter Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this meta ticket https://github.com/ignitionrobotics/sdformat/issues/931

## Summary

Added Particle Emitter Python interface

## Test it

```python
from sdformat import ParticleEmitter

particleEmitter = ParticleEmitter()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pyParticleEmitter.cc [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+#include "pyParticleEmitter.hh"
+
+#include <pybind11/pybind11.h>
+
+#include "sdf/ParserConfig.hh"
+
+#include "sdf/Box.hh"
+#include "sdf/Capsule.hh"
+#include "sdf/Cylinder.hh"
+#include "sdf/Ellipsoid.hh"
```
Comment (azeey) [inline python/src/sdf/pyParticleEmitter.cc:26]: I don't think we need all the other headers
```suggestion
#include "sdf/ParticleEmitter.hh"
```

File: python/src/sdf/pyParticleEmitter.cc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+namespace sdf
+{
+// Inline bracket to help doxygen filtering.
+inline namespace SDF_VERSION_NAMESPACE {
+namespace python
+{
+/////////////////////////////////////////////////
+void defineParticleEmitter(pybind11::object module)
+{
+  // TODO(ahcorde) Add ParticleEmitterShape and SetParticleEmitterShape
```
Comment (azeey) [inline python/src/sdf/pyParticleEmitter.cc:43]: I can't find these functions in the C++ header. Do we need this TODO?

File: python/src/sdf/pyParticleEmitter.cc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+         "is relative to the parent link.")
+    .def("set_pose_relative_to", &sdf::ParticleEmitter::SetPoseRelativeTo,
+         "Set the name of the coordinate frame relative to which this "
+         "object's pose is expressed. An empty value indicates that the frame "
+         "is relative to the parent link.")
+    .def("semantic_pose", &sdf::ParticleEmitter::SemanticPose,
+         "Get SemanticPose object of this object to aid in resolving "
+         "poses.")
+    .def("material", &sdf::ParticleEmitter::Material,
+         pybind11::return_value_policy::reference,
```
Comment (azeey) [inline python/src/sdf/pyParticleEmitter.cc:146]:
```suggestion
         pybind11::return_value_policy::reference_internal,
```

File: python/test/pyParticleEmitter_TEST.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        self.assertFalse(emitter.emitting())
+
+        self.assertAlmostEqual(0.0, emitter.duration())
+        emitter.set_duration(10.0)
+        self.assertAlmostEqual(10.0, emitter.duration())
+
+        self.assertAlmostEqual(5.0, emitter.lifetime())
+        emitter.set_lifetime(22.0)
+        self.assertAlmostEqual(22.0, emitter.lifetime())
+        emitter.set_lifetime(-1.0)
```
Comment (azeey) [inline python/test/pyParticleEmitter_TEST.py:51]: The C++ code has `EXPECT_DOUBLE_EQ(gz::math::MIN_D, emitter.Lifetime());`. Do 
we need to add that?

**Review by azeey — CHANGES_REQUESTED** [review]
Looks great. Just a few minor comments

File: python/test/pyParticleEmitter_TEST.py (hunk 4 above)
Comment (ahcorde) [reply python/test/pyParticleEmitter_TEST.py:51]: https://github.com/gazebosim/sdformat/pull/999/commits/27a4ae4369eb9031c85655ee19b7c0d34b17e199

File: python/src/sdf/pyParticleEmitter.cc (hunk 3 above)
Comment (ahcorde) [reply python/src/sdf/pyParticleEmitter.cc]: https://github.com/gazebosim/sdformat/pull/999/commits/27a4ae4369eb9031c85655ee19b7c0d34b17e199

File: python/src/sdf/pyParticleEmitter.cc (hunk 2 above)
Comment (ahcorde) [reply python/src/sdf/pyParticleEmitter.cc]: https://github.com/gazebosim/sdformat/pull/999/commits/27a4ae4369eb9031c85655ee19b7c0d34b17e199

File: python/src/sdf/pyParticleEmitter.cc (hunk 1 above)
Comment (ahcorde) [reply python/src/sdf/pyParticleEmitter.cc]: https://github.com/gazebosim/sdformat/pull/999/commits/27a4ae4369eb9031c85655ee19b7c0d34b17e199

Comment (azeey) [reply python/src/sdf/pyParticleEmitter.cc]: nit: The headers below `ParticleEmitter` can be removed as well.

**Review by azeey — APPROVED** [review]
One last minor comment, otherwise LGTM!

**Pull request merged**